### PR TITLE
GEOMESA-1783 HBase and Bigtable Improvements

### PR DESCRIPTION
**Pull request by anthonyccri**
* Added Z2 and Z3 pushdown to HBase
* Added BigtableExtendedScan (analog to HBase MultiRowRangeFilter)
* Optimized provisioning of RDDs for HBase and Spark to take advantage
  of pushdown iterators, MultiRowRangeFilter, and BigtableExtendedScan

Signed-off-by: Anthony Fox <anthonyfox@ccri.com>

### CODE REVIEW

File: geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/index/AccumuloIndexAdapter.scala [hunk 1]
```diff
@@ -17,10 +17,12 @@ import org.geotools.factory.Hints
 import org.locationtech.geomesa.accumulo.data.{AccumuloDataStore, AccumuloFeature}
 import org.locationtech.geomesa.accumulo.index.AccumuloIndexAdapter.ScanConfig
 import org.locationtech.geomesa.accumulo.iterators._
+import org.locationtech.geomesa.arrow.vector.ArrowDictionary
```
Comment (jnh5y) [inline geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/index/AccumuloIndexAdapter.scala]: Is this PR blocked by / depending on the Arrow work?

File: geomesa-hbase/geomesa-hbase-datastore/src/main/scala/org/locationtech/geomesa/hbase/data/HBaseDataStoreFactory.scala [hunk 2]
```diff
@@ -53,10 +52,6 @@ class HBaseDataStoreFactory extends DataStoreFactorySpi with LazyLogging {
     // TODO HBase Connections don't seem to be Serializable...deal with it
     val connection = ConnectionParam.lookupOpt[Connection](params).getOrElse(globalConnection)
 
-    val remote = RemoteParam.lookupOpt[Boolean](params)
```
Comment (elahrvivaz) [inline geomesa-hbase/geomesa-hbase-datastore/src/main/scala/org/locationtech/geomesa/hbase/data/HBaseDataStoreFactory.scala]: are we ok with getting rid of non-remote transforms/filters?

File: geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/index/AccumuloIndexAdapter.scala (hunk 1 above)
Comment (elahrvivaz) [inline geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/index/AccumuloIndexAdapter.scala]: arrow was already merged...

**Comment by elahrvivaz** [comment]
replacing with #1503

**Pull request closed without merging**